gateway/tests/integration/routes_test.go

```diff
@@ -50,7 +50,7 @@ func fireRequestWithHeaders(url string, method string, reqBody string, headers m
 func TestGet_Rejected(t *testing.T) {
 	var reqBody string
 	unsupportedMethod := http.MethodHead
-	_, code, err := fireRequest("http://localhost:8080/function/func_echoit", unsupportedMethod, reqBody)
+	_, code, err := fireRequest("http://localhost:8080/function/echoit", unsupportedMethod, reqBody)
```

Comment:
What about a test that works with the handler directly such as for add_metrics_test.go or cors_test.go?

Reply:
This test was changed to get the integration tests to pass, and wasn't directly to do with the feature in this PR

Because of the way that version, commit message and sha are assigned values as part of the build pipeline, I decided to go with an [integration test](https://github.com/openfaas/faas/pull/737/files/32411682d71740f4ffe36359888ef0ff26e0836c#diff-891539b2783190815a650c77acaeefa8). Also `infohandler.go` calls out to the `gateway` so not sure the unit test approach would be appropriate?

Reply:
So for a unit you are testing the logic in the method around error handling/parsing with the data from the upstream call mocked out with canned data instead of a real call as in this instance.